src/CustomElementInternals.js

```diff
@@ -42,6 +42,20 @@ export default class CustomElementInternals {
   }
 
   /**
+   * @param {!Node} node
+   * @return {!CustomElementDefinition|null}
+   */
+  nodeToDefinition(node) {
+    const name = typeof node.__CE_is === 'string' ? node.__CE_is : node.localName;
+    const definition = this.nameToDefinition(name);
+    if (definition && definition.localName === node.localName) {
```

Comment:
It is possible that someone tries to apply a customized builtin element (via `is="my-builtin"`) to a dom element that the customized builtin was never meant to extend. In that case, we should not upgrade the element to be the customized builtin.

This is why there is the `definition.localName === node.localName` check.